R/moduleInstall.R

```diff
@@ -1,12 +1,338 @@
+  if (!dir.exists(moduleLibrary))
+    if (!dir.create(moduleLibrary))
+      stop("failed to create moduleLibrary!")
+
+  moduleLibraryTemp <- file.path(moduleLibrary, ".renv_temp")
+
+  if (!dir.exists(file.path(moduleLibraryTemp, "renv")))
+    dir.create(file.path(moduleLibraryTemp, "renv"), recursive = TRUE)
+
+  setupRenv(moduleLibrary)
+
+  # make renv blind for other libPaths
+  old.lib.loc <- .libPaths()
+  on.exit(assign(".lib.loc", old.lib.loc, envir = environment(.libPaths)))
+  assign(".lib.loc", moduleLibrary, envir = environment(.libPaths))
+
+  # TODO: this is not very efficient because renv::install looks up the remotes on github...
+  # there is a better way but it requires us to mess with renv's internals or to be more explicit about pkgs
+  renv::hydrate(library = moduleLibrary, project = modulePkg)
+  renv::install(project = modulePkg, library = moduleLibrary, prompt = prompt)
+
+  correctlyInstalled <- installModulePkg(modulePkg, moduleLibrary, prompt)
+  if (correctlyInstalled)
+    writeMd5Sums(modulePkg, moduleLibrary)
+
+  if (unlink(moduleLibraryTemp, recursive = TRUE)) # 0/ FALSE for success
+    warning(sprintf("Failed to remove temporary module libary at %s", moduleLibraryTemp))
+
+  return("succes!")
+
+}
+
+installModulePkg <- function(modulePkg, moduleLibrary, prompt = interactive(), moduleInfo = NULL) {
+
+  if (is.null(moduleInfo))
+    moduleInfo <- getModuleInfo(modulePkg)
+  record <- recordFromModule(modulePkg, moduleInfo)
+  renv::install(record, library = moduleLibrary, rebuild = TRUE, prompt = prompt)
+  TRUE
+
+}
 
-installJaspModule <- function(modulePkg, libPathsToUse, moduleLibrary, repos, onlyModPkg) {
+assertValidJASPmodule <- function(modulePkg) {
+
+  if (!file.exists(file.path(modulePkg, "DESCRIPTION")))
+    stop("Your module is missing a 'DESCRIPTION' file!")
+
+  if (!file.exists(file.path(modulePkg, "inst", "Description.qml")))
+    stop("Your module is missing 'inst/Description.qml'!")
+
+  if (!dir.exists(file.path(modulePkg, "R")))
+    stop("Your module is missing an 'R' directory!")
+
+  if (!dir.exists(file.path(modulePkg, "inst", "qml")))
+    stop("Your module is missing the 'inst/qml' directory!")
+
+}
+
+hasRenvLockFile <- function(modulePkg) {
+  return(file.exists(file.path(modulePkg, "renv.lock")))
+}
+
+recordFromModule <- function(modulePkg, moduleInfo) {
+
+  record <- list(list(
+    Package   = moduleInfo[["Package"]],
+    Version   = moduleInfo[["Version"]],
+    Path      = modulePkg,
+    Source    = "Local",
+    Cacheable = TRUE
+  ))
+  names(record) <- moduleInfo[["Package"]]
+
+  return(record)
+}
+
+getModuleInfo <- function(modulePkg) {
+  pkgDescr <- file.path(modulePkg, "DESCRIPTION")
+  if (!file.exists(pkgDescr))
+    stop("Your module contains no DESCRIPTION file")
+
+  return(read.dcf(file.path(modulePkg, "DESCRIPTION"))[1, ])
+
+}
+
+renv_diagnostics_packages_as_df <- function(project) {
+
+  lockfile  <- renv:::renv_diagnostics_packages_lockfile(project)
```

Comment:
perhaps we should add a comment that this function is equivalent to `renv:::renv_diagnostics_packages` except that it returns a data.frame instead of directly printing all information (and returning NULL).